File: app/src/main/java/com/stevenfrew/beatprompter/SongFilter.java
package com.stevenfrew.beatprompter;

import java.util.List;

class SongFilter extends Filter
{
    List<SongFile> mSongs;
    SongFilter(String name, List<SongFile> songs, boolean canSort)
    {
        super(name,canSort);
        mSongs=songs;
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/ui/filter/SongFilter.kt
package com.stevenfrew.beatprompter.ui.filter

import com.stevenfrew.beatprompter.song.SongInfo

abstract class SongFilter internal constructor(
	name: String,
	songs: List<Pair<SongInfo, String?>>,
	canSort: Boolean
) : Filter(name, canSort) {
	val songs = songs.toMutableList()
}